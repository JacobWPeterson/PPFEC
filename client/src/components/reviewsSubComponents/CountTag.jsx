import React from 'react';
import styled from 'styled-components';

const StyledTag = styled.div`
  width: 100%;
  height: 100%;
  font-size: 10px;
  text-align: center;
  border-radius: 6px;
  border: 0.3px solid #424242
  position: absolute
  top: 0;
  left: 50%;
  z-index: 1;
  margin-left: -60px;
`;

const CountTag = (props) => {
  let ratingCount;
  if (props.ratingCount === undefined) {
    ratingCount = 0;
  } else {
    ratingCount = props.ratingCount;
  }
  return (
    <StyledTag>
      <p>
        {ratingCount}
        {' '}
        people voted
      </p>
    </StyledTag>
  );
};

export default CountTag;
