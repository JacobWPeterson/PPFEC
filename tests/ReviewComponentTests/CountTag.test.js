import React from 'react';
import { shallow, mount, render } from 'enzyme';
import CountTag from '../../client/src/components/reviewsSubComponents/CountTag.jsx';

describe('CountTag Component', () => {
  const props = {
    ratingCount: 4,
  };
  let CountTagComponent;
  beforeEach(() => {
    CountTagComponent = mount(<CountTag {...props} debug />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should render correctly', () => {
    expect(CountTagComponent).toMatchSnapshot();
  });
});