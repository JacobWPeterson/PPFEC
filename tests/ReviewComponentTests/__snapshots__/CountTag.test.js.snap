// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountTag Component Should render correctly 1`] = `
<CountTag
  debug={true}
  ratingCount={4}
>
  <CountTag__StyledTag>
    <div
      className="CountTag__StyledTag-sc-169gb2y-0 jPeXft"
    >
      4
       
      votes
    </div>
  </CountTag__StyledTag>
</CountTag>
`;
